// @novu/shared/dist/cjs/consts/preferences/preferences.const.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.DEFAULT_WORKFLOW_PREFERENCES = exports.WORKFLOW_CHANNEL_PREFERENCE_DEFAULT = exports.WORKFLOW_CHANNEL_PREFERENCE_DEFAULT_READ_ONLY = exports.WORKFLOW_CHANNEL_PREFERENCE_DEFAULT_VALUE = void 0;
exports.WORKFLOW_CHANNEL_PREFERENCE_DEFAULT_VALUE = true;
exports.WORKFLOW_CHANNEL_PREFERENCE_DEFAULT_READ_ONLY = false;
exports.WORKFLOW_CHANNEL_PREFERENCE_DEFAULT = {
    enabled: exports.WORKFLOW_CHANNEL_PREFERENCE_DEFAULT_VALUE,
    readOnly: exports.WORKFLOW_CHANNEL_PREFERENCE_DEFAULT_READ_ONLY,
};
exports.DEFAULT_WORKFLOW_PREFERENCES = {
    workflow: exports.WORKFLOW_CHANNEL_PREFERENCE_DEFAULT,
    channels: {
        in_app: exports.WORKFLOW_CHANNEL_PREFERENCE_DEFAULT,
        sms: exports.WORKFLOW_CHANNEL_PREFERENCE_DEFAULT,
        email: exports.WORKFLOW_CHANNEL_PREFERENCE_DEFAULT,
        push: exports.WORKFLOW_CHANNEL_PREFERENCE_DEFAULT,
        chat: exports.WORKFLOW_CHANNEL_PREFERENCE_DEFAULT,
    },
};
